### ISSUE TITLE
Multiple widgets support

### ISSUE THREAD

**Issue by ruwiss** [ABausG/home_widget#189]
Please add a property like widgetId so we can customize multiple widgets.

**Comment by ruwiss** [comment]
I couldn't understand there are probably different ways to do it.
When the widget is first added, the application should be opened to configure it.

**Comment by ABausG** [comment]
What do you mean with adding a widgetID? Multiple are already supported. You can use the widget Ids yourself and say e.g:

```dart
HomeWidget.saveData('widget1_title', 'Title1');
HomeWidget.saveData('widget2_title', 'Title2');

HomeWidget.updateWidget(androidName: 'Widget1');
HomeWidget.updateWidget(androidName: 'Widget2');
```

I don't have a dedicated 'Edit Widget' Functionallity right now (possibly in the future) However you can detect if your app was opened via a click on the Widget so should be doable that way

**Comment by ruwiss** [comment]
But what I'm really trying to say 
When I create more than one view with widget1 id, I can assign a custom id to them. This way we can customize multiple widgets with a single view. I don't understand how to do that

**Comment by thanglq1** [comment]
> I created 1 countdown widget, let's call it CustomWidget. I have a countdown / planning app In my app, users can create countdown(s). If they add a countdown on the home screen, only 1 countdown works on the CustomWidget. But if I want to run more than one countdown, I can't do that. Because I can only give 1 countdown to 1 home widget. So I need to be able to add id to my CustomWidget view. Like in the package named pub.dev app_widget.

Hi @ruwiss ruwis. Can you share your code? I have the same task as well. I need add more "note" on home screen.

**Comment by ruwiss** [comment]
> Hi @ruwiss ruwis. Can you share your code? I have the same task as well. I need add more "note" on home screen.

I'm sorry but I couldn't code it the way we wanted

**Comment by Codestian** [comment]
> What do you mean with adding a widgetID? Multiple are already supported. You can use the widget Ids yourself and say e.g:
> 
> ```dart
> HomeWidget.saveData('widget1_title', 'Title1');
> HomeWidget.saveData('widget2_title', 'Title2');
> 
> HomeWidget.updateWidget(androidName: 'Widget1');
> HomeWidget.updateWidget(androidName: 'Widget2');
> ```
> 
> I don't have a dedicated 'Edit Widget' Functionallity right now (possibly in the future) However you can detect if your app was opened via a click on the Widget so should be doable that way

I think the issue is what happens if the user creates an indefinite amount of widgets, creating and removing them many times. Since there keys in the shared preferences will continue to pull up and will not be disposed.

**Comment by ABausG** [comment]
You can now use `getInstalledWidgets` to check how many/which widgets a user has installed to clean up the data from flutter. Available in Version 0.5.0
